### ISSUE TITLE
error: The name `Settings' does not exist in the context of `Granite' (granite)

### ISSUE THREAD

**Issue by GitEin11** [donadigo/appeditor#130]
OS: Ubuntu 20.04.6 LTS x86_64 

sudo ninja install

[1/24] Compiling Vala source Config.vala ../src/Application.vala...conChooserDialog.vala ../src/IconRow.vala ../src/Constants.vala.
FAILED: com.github.donadigo.appeditor@exe/build/Config.c com.github.donadigo.appeditor@exe/src/Application.c com.github.donadigo.appeditor@exe/src/MainWindow.c com.github.donadigo.appeditor@exe/src/AppSourceList.c com.github.donadigo.appeditor@exe/src/AppItem.c com.github.donadigo.appeditor@exe/src/CategoryItem.c com.github.donadigo.appeditor@exe/src/IconButton.c com.github.donadigo.appeditor@exe/src/SettingBox.c com.github.donadigo.appeditor@exe/src/PersistentPlaceholderEntry.c com.github.donadigo.appeditor@exe/src/FieldEntry.c com.github.donadigo.appeditor@exe/src/CategoryComboBox.c com.github.donadigo.appeditor@exe/src/AppInfoViewStack.c com.github.donadigo.appeditor@exe/src/AppInfoView.c com.github.donadigo.appeditor@exe/src/AppDirectoryScanner.c com.github.donadigo.appeditor@exe/src/DesktopApp.c com.github.donadigo.appeditor@exe/src/DesktopAppManager.c com.github.donadigo.appeditor@exe/src/AppInfoViewSaver.c com.github.donadigo.appeditor@exe/src/IconListBox.c com.github.donadigo.appeditor@exe/src/IconChooserDialog.c com.github.donadigo.appeditor@exe/src/IconRow.c com.github.donadigo.appeditor@exe/src/Constants.c 
valac -C --debug --debug --define HAS_VALA040 --pkg posix --pkg posix --pkg posix --pkg gee-0.8 --pkg gtk+-3.0 --pkg granite --color=always --directory com.github.donadigo.appeditor@exe --basedir ../ --gresources=../data/com.github.donadigo.appeditor.gresource.xml Config.vala ../src/Application.vala ../src/MainWindow.vala ../src/AppSourceList.vala ../src/AppItem.vala ../src/CategoryItem.vala ../src/IconButton.vala ../src/SettingBox.vala ../src/PersistentPlaceholderEntry.vala ../src/FieldEntry.vala ../src/CategoryComboBox.vala ../src/AppInfoViewStack.vala ../src/AppInfoView.vala ../src/AppDirectoryScanner.vala ../src/DesktopApp.vala ../src/DesktopAppManager.vala ../src/AppInfoViewSaver.vala ../src/IconListBox.vala ../src/IconChooserDialog.vala ../src/IconRow.vala ../src/Constants.vala
../src/Application.vala:61.32-61.47: error: The name `Settings' does not exist in the context of `Granite' (granite)
        var granite_settings = Granite.Settings.get_default ();
                               ^^^^^^^^^^^^^^^^
../src/Application.vala:64.99-64.114: error: The name `Settings' does not exist in the context of `Granite' (granite)
        gtk_settings.gtk_application_prefer_dark_theme = granite_settings.prefers_color_scheme == Granite.Settings.ColorScheme.DARK;
                                                                                                  ^^^^^^^^^^^^^^^^
Compilation failed: 2 error(s), 0 warning(s)
ninja: build stopped: subcommand failed.

**Comment by ryonakano** [comment]
This is because `Granite.Settings` is only available since [5.4.0](https://github.com/elementary/granite/releases/tag/5.4.0), while the available grantie on universe of Ubuntu 20.04 is 5.3.0.

Consider building and installing the granite 5.4.0 (or above) and then try building of AppEditor, or use newer Ubuntu versions.

**Comment by GitEin11** [comment]
> This is because `Granite.Settings` is only available since [5.4.0](https://github.com/elementary/granite/releases/tag/5.4.0), while the available grantie on universe of Ubuntu 20.04 is 5.3.0.
> 
> Consider building and installing the granite 5.4.0 (or above) and then try building of AppEditor, or use newer Ubuntu versions.


thanks I was able to build granite 5.4.0

```
sudo apt install gobject-introspection libgirepository1.0-dev


then downloading [5.4.0](https://github.com/elementary/granite/releases/tag/5.4.0)

meson build --prefix=/usr
cd build
ninja
ninja install

finally I was able to build and install appeditor :)
```
